src/test/update-course-links.test.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const path = require('path');
+const test = require('ava');
+const fetchMock = require('fetch-mock');
+const parser = require('parse5');
+
+const CourseLinkUpdater = require('../update-course-links');
+
+const ContentPath = path.join(__dirname, 'content');
+const ManifestPath = path.join(__dirname, 'content/manifest.json');
+
+const MockValence = {
+	createAuthenticatedUrl(url) {
+		return url;
+	}
+};
+
+const OrgUnit = {
+	Identifier: '123',
+	Name: 'Org Unit',
+	Path: '/content/course123/'
+};
+
+test('_findInManifest finds topic in module', t => {
+	const manifest = {
+		modules: [{
+			title: 'module1',
+			children: [{
+				title: 'topic1'
+			}]
+		}]
+	};
+
+	class MockModuleProcessor { }
+
+	const fetch = fetchMock.sandbox();
+
+	const processor = new CourseLinkUpdater({ contentDirectory: ContentPath, manifestPath: ManifestPath }, MockValence, fetch, MockModuleProcessor);
+
+	const result = processor._findInManifest(x => x.title === 'topic1', manifest);
+
+	t.deepEqual(result, {
+		title: 'topic1'
+	});
+});
+
+test('_findInManifest finds topic in submodule', t => {
+	const manifest = {
+		modules: [{
+			title: 'module1',
+			children: [{
+				title: 'module2',
+				children: [{
+					title: 'topic1'
+				}]
+			}]
+		}]
+	};
+
+	class MockModuleProcessor { }
+
+	const fetch = fetchMock.sandbox();
+
+	const processor = new CourseLinkUpdater({ contentDirectory: ContentPath, manifestPath: ManifestPath }, MockValence, fetch, MockModuleProcessor);
+
+	const result = processor._findInManifest(x => x.title === 'topic1', manifest);
+
+	t.deepEqual(result, {
+		title: 'topic1'
+	});
+});
+
```

Comment:
We might need to talk about the expectations of titles being unique, but I think a test verifying that the correct `topic1` is returned would be valuable.   It's too easy to have an `introduction` topic in two different modules.